tracee-rules/types/types.go

```diff
@@ -49,3 +49,8 @@ type Finding struct {
 	Context   Event
 	Signature Signature
 }
+
+type FindingWithMetadata struct {
+	Finding
+	SignatureMetadata
+}
```

Comment:
@simar7 what do you think on changing Finding type to include the SignatureMetadata instead of the Signature interface? @yanivagman also curios what you think

Reply:
Signature interface has GetMetadata() method - why not using it when needed?

Reply:
Yaniv and I discussed this offline, we should replace signature field with signaturemetadata in the finding. This will simplify many cases without sacrificing much.

Reply:
Should we do that as part of this PR? I'm happy to do it just feel it's unrelated to this PR as modifying an existing interface requires all users of it (example signatures in this case) to be changed as well.

Reply:
whatever you prefer

Reply:
I'll draft a new PR

Reply:
https://github.com/aquasecurity/tracee/pull/646